app/routes/city-council-districts.$cityCouncilDistrictId.capital-projects.tsx

```diff
@@ -0,0 +1,78 @@
+import { LoaderFunctionArgs, json } from "@remix-run/node";
+import { findCapitalProjectsByCityCouncilId, findAgencies } from "../gen";
+import { useLoaderData, useNavigate } from "@remix-run/react";
+import { CapitalProjectsAccordionPanel } from "../components/CapitalProjectsList";
+import { Box, Hide, Show, Text } from "@nycplanning/streetscape";
+import { CapitalProjectsDrawer } from "~/components/CapitalProjectsList/CapitalProjectsDrawer";
+
+export async function loader({ request, params }: LoaderFunctionArgs) {
+  const url = new URL(request.url);
+  const path = url.pathname;
+  const agenciesResponse = await findAgencies({
+    baseURL: `${import.meta.env.VITE_ZONING_API_URL}/api`,
+  });
+  const limitParam = url.searchParams.get("limit");
+  const offsetParam = url.searchParams.get("offset");
+  let limit;
+  let offset;
+  if (limitParam === null) {
+    limit = 20;
+  } else {
+    limit = parseInt(limitParam);
+  }
+  if (offsetParam === null) {
+    offset = 0;
+  } else {
+    offset = parseInt(offsetParam);
+  }
+
+  const { cityCouncilDistrictId } = params;
+  if (cityCouncilDistrictId === undefined) {
+    throw json("Bad Request", { status: 400 });
+  }
+
+  const projectsByCityCouncilDistrictResponse =
+    await findCapitalProjectsByCityCouncilId(
+      cityCouncilDistrictId,
+      {
+        limit: 7,
+        offset,
+      },
+      {
+        baseURL: `${import.meta.env.VITE_ZONING_API_URL}/api`,
+      },
+    );
+
+  return {projectsByCityCouncilDistrictResponse, agencies: agenciesResponse.agencies, limit, offset, path, cityCouncilDistrictId};
+}
+
+export default function CapitalProjectsByCityCouncilDistrict() {
+    const {projectsByCityCouncilDistrictResponse, agencies, cityCouncilDistrictId, limit, path, offset} = useLoaderData<typeof loader>();
+
+    return (
+        <>
+        <Show above='sm'>
+            <CapitalProjectsAccordionPanel
+                capitalProjects={projectsByCityCouncilDistrictResponse.capitalProjects}
+                limit={limit}
```

Comment:
I do like that we pass the limit and offset to the child components, rather than using the hooks within the children to access these values. In my opinion, only the `route` level components should be aware that they're on a specific path. Every child component should function independently of the route they're on. Passing the limit and offset parameters facilitates this.